server/controller/semester.controller.js

```diff
@@ -34,36 +40,49 @@ semesterController.post(
   passport.authenticate('jwt', { session: false }),
   validateCreateSemester,
   async (req, res) => {
-    const errorsAfterValidation = validationResult(req);
-    if (!errorsAfterValidation.isEmpty()) {
-      res.status(400).json({
-        code: 400,
-        errors: errorsAfterValidation.mapped(),
-      });
-    } else {
+    validationHandler(req, res, async () => {
       try {
-        const { term, year, courses } = req.body;
-        const semester = await Semester.findOne({ term, year, courses });
+        const semester = await Semester.findOne(req.body);
```

Comment:
Inconsistent POST api design between the course and semester controller. I would keep how we handle errors and callbacks the same, so either as above (in semester controller) or below (in course controller):

await Course.findOne({ school, code }, (err, foundCourse) => {
          if (!foundCourse)
            courseController
              .createCourse({ school, code }, 'NEW_COURSE')
              .then(createdCourse => {
                return res.status(200).json(createdCourse);
              });